packages/core/src/validation.ts

```diff
@@ -0,0 +1,41 @@
+/********************************************************************************
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the W3C Software Notice and
+ * Document License (2015-05-13) which is available at
+ * https://www.w3.org/Consortium/Legal/2015/copyright-software-and-document.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR W3C-20150513
+ ********************************************************************************/
+
+export function isThingDescription(input: unknown): input is WoT.ThingDescription {
+    // TODO: Improve validation function
```

Comment:
Does it make sense to also have proper validation based on the JSON schema (AJV) validation like we do for TMs.
https://github.com/eclipse-thingweb/node-wot/blob/0fd864b04c0badca2b8dbb90f01144614c36c37d/packages/td-tools/src/thing-model-helpers.ts#L154

Anyhow, definitely not in this PR and I am not even sure whether it makes sense int this case since we should be relaxed when consuming a TD and as strict as possible when creating a TD.

Reply:
Thank you for your feedback! I actually had already started to change the validation approach so that it uses the JSON schema definition for validation and also added it to the PR now in the latest commits :) Hope that this is still fine